### ISSUE TITLE
3.10.0.1 breaks CI in importlib_metadata

### ISSUE THREAD

**Issue by jaraco** [python/typing#865]
In https://github.com/python/importlib_metadata/issues/350, I observed that the tests on Python 3.10 started failing. Looking at a diff of the dependencies, I can see that the issue is brought about by the introduction of the recently released typing-extensions==3.10.0.1.

Error message is:

```
    class _ConcatenateGenericAlias(list):
    
        # Trick Generic into looking into this for __parameters__.
        if PEP_560:
>           __class__ = _GenericAlias
E           NameError: name '_GenericAlias' is not defined

.tox/python/lib/python3.10/site-packages/typing_extensions.py:2394: NameError
```

I suspect this issue will have widespread impact on environments running under Python 3.10.

**Comment by JelleZijlstra** [comment]
Thanks for bringing this up! I can release a new version as soon as this is fixed, but we should probably get CI back up first (#864).

**Comment by Fidget-Spinner** [comment]
@jaraco thanks for highlighting this to us and sorry about that. I'll try to fix this ASAP.

**Comment by sobolevn** [comment]
Fixed in #869

**Comment by Fidget-Spinner** [comment]
Yep. Thanks again @sobolevn!

**Comment by JelleZijlstra** [comment]
The CI PR might take a while longer so I'm going to try running tests locally myself and then release 3.10.0.2 with the fix.

**Comment by srittau** [comment]
@JelleZijlstra Feel free to mehr the CI PR. I think it's "good enough" and can be cleaned up later

**Comment by JelleZijlstra** [comment]
We need to wait for the CLA to land.

**Comment by srittau** [comment]
> We need to wait for the CLA to land.

Ah, ok.

**Comment by Fidget-Spinner** [comment]
I'm awfully confused why plain `_GenericAlias` passed in 3.9 but failed in 3.10 (also I could've sworn that I tested it locally on 3.10 and it passed 😞.

**Comment by mgorny** [comment]
Note that there were some last-minutes changes in `typing` in Python 3.10.0rc1, so it might have been that if you were testing on b4.

**Comment by AMDmi3** [comment]
Judging by my runs, it's similarly broken everywhere from 3.10.0-b2 to 3.10.0-rc1.

**Comment by Fidget-Spinner** [comment]
Nevermind, it seems that I'd made a mistake -- I just tested this on 3.10a7 and it didn't pass there either. I'm really sorry everyone.

Hopefully our new CI with 3.10 runs will catch such things in the future.

**Comment by JelleZijlstra** [comment]
Update: I just released 3.10.0.2, which passes tests under 3.10.0rc1.
